3303Elevator/src/app/ElevatorSubsystem/ElevatorSubsystem.java

```diff
@@ -17,83 +22,41 @@
  * */
 public class ElevatorSubsystem implements Runnable{
 
-	/**
-	 * Name for the elevator
-	 * */
 	private String name;
-
-	/**
-	 * Elevator the subsystem controls
-	 * */
-	private Elevator elevator;
-
-	/**
-	 * The scheduler the elevator needs to communicate with
-	 * */
-	private Scheduler scheduler;
-
-	/**
-	 * Maximum number of floors
-	 */
-	private int maxFloorCount;
-	
-	/**
-	 * Logger to track what happens
-	 */
+	private ArrayList<Elevator> elevators;
+	private InetSocketAddress schedulerAddr;
+	private int maxFloor;
 	private Logger logger;
-	
 	private TimeManagementSystem tms;
+	
 	/**
 	 * Constructor used to create elevator subsystem
 	 *
 	 * @param scheduler 	 the scheduler used to communication
 	 * */
-	public ElevatorSubsystem(Scheduler scheduler, int maxFloorCount, Logger logger, TimeManagementSystem tms){
-		this.maxFloorCount = maxFloorCount;
+	public ElevatorSubsystem(InetSocketAddress schedulerAddr, int numElevators, int maxFloor, Logger logger, TimeManagementSystem tms){
 		this.name = Thread.currentThread().getName();
-		this.elevator = new Elevator(maxFloorCount, tms);
-		this.scheduler = scheduler;
-		this.logger = logger;
+		this.maxFloor = maxFloor;
+		
+		elevators = new ArrayList<Elevator>();
+		for(int i = 0; i < numElevators; i++) {
```

Comment:
Consider having elevator ID number as a param to elevator constructor. I feel like subsys should tell elevators what number they are instead of elevators intrinsically knowing what number they are with the shared static variable